api/imapsync/update

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/perl
+#
+#
+# Copyright (C) 2020 Nethesis S.r.l.
+# http://www.nethesis.it - nethserver@nethesis.it
+#
+# This script is part of NethServer.
+#
+# NethServer is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License,
+# or any later version.
+#
+# NethServer is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with NethServer.  If not, see COPYING.
+#
+
+use strict;
+use esmith::ConfigDB;
+use esmith::event;
+use JSON;
+use NethServer::ApiTools;
+
+my $ret;
+my $db = esmith::ConfigDB->open('imapsync');
+my $input = NethServer::ApiTools::readInput();
+my $cmd = $input->{'action'};
+
+if ($cmd eq 'update') {
+    foreach (qw (
+        hostname
+        username
+        DeleteDestination
+        Port
+        Security
+    )) {
+        $db->set_prop($input->{'name'}, $_, $input->{$_}, type => 'imapsync');
+    }
+
+    umask 077;
+    open(FH, '>', "/var/lib/nethserver/secrets/imapsync-$input->{'name'}");
+    print FH $input->{'password'};
+    close FH;
```

Comment:
Decide if you want to use NethServer::Password (and `/var/lib/nethserver/secrets`) or read/write/delete password files directly. Please, do not mix the two things.